### DISCUSSION TITLE
Thank you

### DISCUSSION THREAD

**Discussion by jbeltran73-2** [linux-surface/linux-surface#957] [General]
Just wanted to say thank you for such an amazing progress. I love my surface, but better with Linux. Awesome job, everyone. Thank you!

**Comment by Drzt1260** [comment]
Yes i want to join this thank you threat and want to leave a freedback for people who are not sure which distribution they should use.

I use the Surface Pro 5th Gen 2017 with Ubuntu / Mint 21.2 Vicotria.

And its very easy to set up. Once i followed the instrucations it is just perfect. Everything is working very nice and smooth (Touchscreen, also in Firefox, Sound, Keyboard + Touchpad) All update functions seem to work properly and in my opinion the performance is even better than with Windows 10. Yes I'm pretty sure, that its running better than on their own OS, but I am not gaming at all. I've installed dual boot OS, so i could use Windows... But since i have Mint installed about an year ago I don't use it anymore at all.

There are just to exceptions I have to add. I cant check the energydrain, because on this SF5 the batterie is absolutely broken. If you disconnect it from the power its almost empty after 20minutes... also on Windows. Too bad that they failed so hard during the constuction of this device, its not designed for repairing.

And I don't cant confirm that the camera is working. I will never use it and in this case I don't want it to be functional. 

So thank you all! You did a very good job and still there are kernel updates in higher frequency than the Manufacture. BTW MS Surface 5 Pro will not be officially supported for Windows 11. Good job, you don't even support a business device of your own house.

So don't hesitate to try a Linux OS on Surfaces. 

System-info attached.

```
System:
  Kernel: 6.4.7-surface x86_64 bits: 64 compiler: gcc v: 9.4.0 Desktop: Cinnamon 5.8.4
    tk: GTK 3.24.33 wm: muffin dm: LightDM Distro: Linux Mint 21.2 Victoria base: Ubuntu 22.04 jammy
Machine:
  Type: Laptop System: Microsoft product: Surface Pro
  
  Mobo: Microsoft model: Surface Pro serial: <superuser required> UEFI: Microsoft v: 238.167.768
    date: 05.07.2014
Battery:
  ID-1: BAT1 charge: 23.1 Wh (100.0%) condition: 23.1/45.0 Wh (51.3%) volts: 8.5 min: 7.6
    model: SMP M1009169 serial: <filter> status: Full
CPU:
  Info: dual core model: Intel Core i7-7660U bits: 64 type: MT MCP arch: Amber/Kaby Lake
    note: check rev: 9 cache: L1: 128 KiB L2: 512 KiB L3: 4 MiB
  Speed (MHz): avg: 2050 high: 2500 min/max: 400/4000 cores: 1: 2500 2: 2500 3: 2500 4: 700
    bogomips: 19999
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx
Graphics:
  Device-1: Intel Iris Plus Graphics 640 vendor: Microsoft driver: i915 v: kernel ports:
    active: eDP-1 empty: DP-1, DP-2, HDMI-A-1, HDMI-A-2 bus-ID: 00:02.0 chip-ID: 8086:5926
  Display: x11 server: X.Org v: 1.21.1.4 driver: X: loaded: modesetting unloaded: fbdev,vesa
    gpu: i915 display-ID: :0 screens: 1
  Screen-1: 0 s-res: 2736x1824 s-dpi: 96
  Monitor-1: eDP-1 model: LG res: 2736x1824 dpi: 267 diag: 312mm (12.3")
  OpenGL: renderer: Mesa Intel Iris Plus Graphics 640 (Kaby Lake GT3e) (KBL GT3)
    v: 4.6 Mesa 23.0.4-0ubuntu1~22.04.1 direct render: Yes
Audio:
  Device-1: Intel Sunrise Point-LP HD Audio driver: snd_hda_intel v: kernel bus-ID: 00:1f.3
    chip-ID: 8086:9d71
  Sound Server-1: ALSA v: k6.4.7-surface running: yes
  Sound Server-2: PulseAudio v: 15.99.1 running: yes
  Sound Server-3: PipeWire v: 0.3.48 running: yes
Network:
  Device-1: Marvell 88W8897 [AVASTAR] 802.11ac Wireless driver: mwifiex_pcie v: 1.0 pcie:
    speed: 2.5 GT/s lanes: 1 bus-ID: 01:00.0 chip-ID: 11ab:2b38
  IF: wlp1s0 state: up mac: <filter>
Bluetooth:
  Device-1: Marvell Bluetooth and Wireless LAN Composite type: USB driver: btusb v: 0.8
    bus-ID: 1-5:4 chip-ID: 1286:204c
  Report: hciconfig ID: hci0 rfk-id: 0 state: up address: <filter> bt-v: 2.1 lmp-v: 4.2
    sub-v: 7015
Drives:
  Local Storage: total: 238.47 GiB used: 55.96 GiB (23.5%)
  ID-1: /dev/nvme0n1 vendor: Samsung model: KUS030202M-B000 size: 238.47 GiB speed: 15.8 Gb/s
    lanes: 2 serial: <filter> temp: 43.9 C
Partition:
  ID-1: / size: 45.53 GiB used: 33.29 GiB (73.1%) fs: ext4 dev: /dev/nvme0n1p6
  ID-2: /boot/efi size: 96 MiB used: 31.4 MiB (32.7%) fs: vfat dev: /dev/nvme0n1p1
Swap:
  ID-1: swap-1 type: partition size: 5.59 GiB used: 0 KiB (0.0%) priority: -2 dev: /dev/nvme0n1p5
Sensors:
  System Temperatures: cpu: 45.0 C pch: 40.0 C mobo: N/A
  Fan Speeds (RPM): N/A
Repos:
  Packages: 2312 apt: 2292 flatpak: 10 snap: 10
  No active apt repos in: /etc/apt/sources.list
  Active apt repos in: /etc/apt/sources.list.d/linux-surface.list
    1: deb [arch=amd64] https: //pkg.surfacelinux.com/debian release main
  Active apt repos in: /etc/apt/sources.list.d/nordvpn.list
    1: deb https: //repo.nordvpn.com/deb/nordvpn/debian stable main
  Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list
    1: deb http: //packages.linuxmint.com victoria main upstream import backport
    2: deb http: //archive.ubuntu.com/ubuntu jammy main restricted universe multiverse
    3: deb http: //archive.ubuntu.com/ubuntu jammy-updates main restricted universe multiverse
    4: deb http: //archive.ubuntu.com/ubuntu jammy-backports main restricted universe multiverse
    5: deb http: //security.ubuntu.com/ubuntu/ jammy-security main restricted universe multiverse
  Active apt repos in: /etc/apt/sources.list.d/spotify.list
    1: deb http: //repository.spotify.com stable non-free
  No active apt repos in: /etc/apt/sources.list.d/extrepo_surface-linux.sources
Info:
  Processes: 239 Uptime: 12m Memory: 7.67 GiB used: 4.1 GiB (53.5%) Init: systemd v: 249
  runlevel: 5 Compilers: gcc: 11.4.0 alt: 11/12 Client: Cinnamon v: 5.8.4 inxi: 3.3.13
```

**Comment by dylanetaft** [comment]
Replying, thank you.  Surface on Linux is so well supported thanks to the efforts of the devs here.  It is rare to find new hardware so well supported.

**Comment by 4iMh9CawHdyu** [comment]
I second this. Surface is sooooo much better with linux. Even Surface Pro 7 which isn't as well implemented yet as the other models.

**Comment by FireNWater** [comment]
Successful Surface Pro 4 install here.  Currently have GRUB dual-boot set up, but will probably be kaputting Windows in a couple months.  Camera not working yet, but that's OK.  Touch, WiFi, and Bluetooth are all working well.  This laptop was getting laggy on Windows 10 and Win 11 wasn't available as an upgrade.  I will now be able to get a few years more out of this perfectly useable laptop.

[new to linux] I assume now that I've added this repository to my Surface Pro 4, it will receive all new updates to this repository?  Thanks!!

**Reply by qzed** [reply]
> [new to linux] I assume now that I've added this repository to my Surface Pro 4, it will receive all new updates to this repository?

Yes. Updates will be installed alongside the normal updates shipped by your distribution through the package manager.

**Reply by FireNWater** [reply]
Booyah!!!

**Comment by devtribun** [comment]
Very good job. A big thank you to all the people who contributed to free the Surface Hardware from the Windows-cellar. I want to encourage everyone to risk this step.

I did not try to setup a dual-boot-device, because I wanted to get rid of all this MS stuff after all these years. Thank you MS for stopping the support of Windows 10 to encourage me to quit your empire.

I just installed Mint 22 Cinnamon using Ventoy in Grep2 mode (highly recommended!) on my Surface Pro 4, using all the help and support provided by this project. I'm a nearly noobe to Linux with very poor terminal knowledge. But my SP4 is running now smooth and fast.

First I tried to install Fedora KDE, but I could not get some hardware to work. The dealbreaker was the broken WiFi.

With Mint it took me several hours, too, to get everything configured. But finally almost all hardware was supported out of the box (but not the cameras).  Using the device as a tablet (without hardware keyboard) was and is still a challenge, a lot of try and error.

I managed to get libcamera installed, which enabled the two cameras - as the terminal reported. But until now I did not get the cameras working in an app. This is no big deal for me, because this device is ment for other usecases.

I'm an adult with adult finger sizes. Scrolling in the menu is still very hard, because the scrollbars are so tiny. It seems the settings of the scrollbars unter 'theme' do not affect the menu. I had to do some tweaks in Firefox under about:config to increase the scrollbars and some other browser-settings to match with the SP4 under Linux. I did not needed these tweaks unter Windows.

I installed a different file-manager (Dolphin). I tried the KDE Krusader, but had too many problems to get it configured. 

There is still some work to do. But all in all I have a fast, stable device for the things it is meant for.

Next stop: Switching all other devices from Windows to Linux. I started with the most challenging device, the SP4, to get prepared for the devices I need for work.

Thank you all again for all your work and the hours of your lifetime you spent on this project.

Peace.

**Comment by stuartm9999** [comment]
Thanks from me too - using my Surface Pro 4 - lovely device - full time for work with Ubuntu 25.10 "Questing Quokka".

Cameras are working and I can use them in teams PWA using the "GStreamer Loopback" - whatever that is it works!

New lease of life!

Stuart
